hydromt/workflows/mesh.py

```diff
@@ -0,0 +1,237 @@
+"""Implementation for mesh based workflows."""
+import logging
+from typing import Dict, Optional, Union
+
+import geopandas as gpd
+import xugrid as xu
+from pyproj import CRS
+from shapely.geometry import box
+from xugrid.ugrid import conventions
+
+from .. import gis_utils
+from ..raster import GEO_MAP_COORD
+from .basin_mask import parse_region
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "create_mesh2d",
+    "rename_mesh",
+]
+
+
+def create_mesh2d(
+    region: Dict,
+    res: Optional[float] = None,
+    crs: Optional[int] = None,
+    align: bool = True,
+    logger=logger,
+) -> xu.UgridDataset:
+    """
+    Create an 2D unstructured mesh or reads an existing 2D mesh.
+
+    Grids are read according to UGRID conventions. An 2D unstructured mesh
+    will be created as 2D rectangular grid from a geometry (geom_fn) or bbox.
+    If an existing 2D mesh is given, then no new mesh will be generated but an extent
+    can be extracted using the `bounds` argument of region.
+
+    Note Only existing meshed with only 2D grid can be read.
+
+    Adds/Updates model layers:
+
+    * **grid_name** mesh topology: add grid_name 2D topology to mesh object
+
+    Parameters
+    ----------
+    region : dict
+        Dictionary describing region of interest, bounds can be provided
+        for type 'mesh'. In case of 'mesh', if the file includes several
+        grids, the specific 2D grid can be selected using the 'grid_name' argument.
+        CRS for 'bbox' and 'bounds' should be 4326; e.g.:
+
+        * {'bbox': [xmin, ymin, xmax, ymax]}
+
+        * {'geom': 'path/to/polygon_geometry'}
+
+        * {'mesh': 'path/to/2dmesh_file'}
+
+        * {'mesh': 'path/to/2dmesh_file', 'grid_name': 'mesh2d', 'bounds': [xmin, ymin, xmax, ymax]}
+    res: float
+        Resolution used to generate 2D mesh [unit of the CRS], required if region
+        is not based on 'mesh'.
+    crs : EPSG code, int, optional
+        Optional EPSG code of the model or "utm" to let hydromt find the closest
+        projected CRS. If None using the one from region, and else 4326.
+    align : bool, optional
+        Align the mesh to the resolution, by default True.
+
+    Returns
+    -------
+    mesh2d : xu.UgridDataset
+        Generated mesh2d.
+    """  # noqa: E501
+    kind, region = parse_region(region, logger=logger)
+    if kind != "mesh":
+        if not isinstance(res, (int, float)):
+            raise ValueError("res argument required for kind 'bbox', 'geom'")
```

Comment:
for clarity I think it'd be good to show the type that was actually passed.

Reply:
True. but here I expect more that people forget to set the argument and the default None is returned, so I could also have put the check to if res is not None. So I think the error message should be okay or clearer this way.